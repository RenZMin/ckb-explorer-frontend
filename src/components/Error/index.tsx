import PCDataNotFoundImage from '../../assets/pc_data_not_found.png'
import MobileDataNotFoundImage from '../../assets/mobile_data_not_found.png'
import { isMobile } from '../../utils/screen'
import { ErrorPanel } from './styled'

export default () => (
  <ErrorPanel>
    <img alt="data not found" src={isMobile() ? MobileDataNotFoundImage : PCDataNotFoundImage} />
  </ErrorPanel>
)
