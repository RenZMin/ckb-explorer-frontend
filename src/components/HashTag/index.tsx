import { TagPanel } from './styled'

export default ({ content, category = 'lock' }: { content: string; category?: 'lock' | 'type' }) => (
  <TagPanel isLock={category === 'lock'} length={content.length}>
    <span>{content}</span>
  </TagPanel>
)
