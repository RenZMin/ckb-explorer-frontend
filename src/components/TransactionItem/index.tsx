import React, { ReactNode } from 'react'
import { Link } from 'react-router-dom'
import RightArrowIcon from '../../assets/input_arrow_output.png'
import DownArrowIcon from '../../assets/input_arrow_output_down.png'
import { parseDate } from '../../utils/date'
import { localeNumberString } from '../../utils/number'
import { isMobile, isScreenSmallerThan1200 } from '../../utils/screen'
import { adaptPCEllipsis, adaptMobileEllipsis } from '../../utils/string'
import TransactionCell from './TransactionItemCell'
import TransactionCellList from './TransactionItemCellList'
import TransactionIncome from './TransactionIncome'
import { FullPanel, TransactionHashBlockPanel, TransactionCellPanel, TransactionPanel } from './styled'
import i18n from '../../utils/i18n'
import { CellType } from '../../utils/const'

export interface CircleCorner {
  top?: boolean
  bottom?: boolean
}

const TransactionItem = ({
  transaction,
  address,
  isBlock = false,
  titleCard,
  circleCorner = {
    top: false,
    bottom: false,
  },
}: {
  transaction: State.Transaction
  address?: string
  isBlock?: boolean
  titleCard?: ReactNode | null
  circleCorner?: CircleCorner
}) => {
  const txHashMobile = adaptMobileEllipsis(transaction.transactionHash, 10)
  const txHashPC = adaptPCEllipsis(transaction.transactionHash, 17, 28)

  return (
    <TransactionPanel id={isBlock && transaction.isCellbase ? 'cellbase' : ''} circleCorner={circleCorner}>
      {titleCard}
      <TransactionHashBlockPanel>
        <div className="transaction_item__content">
          <Link to={`/transaction/${transaction.transactionHash}`}>
            <span className="transaction_item__hash monospace">{isMobile() ? txHashMobile : txHashPC}</span>
          </Link>
          {!isBlock && (
            <div className="transaction_item__block">
              {`(${i18n.t('block.block')} ${localeNumberString(transaction.blockNumber)})  ${parseDate(
                transaction.blockTimestamp,
              )}`}
            </div>
          )}
        </div>
      </TransactionHashBlockPanel>
      <TransactionCellPanel>
        <div className="transaction_item__input">
          <TransactionCellList
            cells={transaction.displayInputs}
            transaction={transaction}
            render={cell => <TransactionCell cell={cell} address={address} cellType={CellType.Input} key={cell.id} />}
          />
        </div>
        <img src={isScreenSmallerThan1200() ? DownArrowIcon : RightArrowIcon} alt="input and output" />
        <div className="transaction_item__output">
          {transaction.displayOutputs && transaction.displayOutputs.length !== 0 ? (
            <TransactionCellList
              cells={transaction.displayOutputs}
              transaction={transaction}
              render={cell => (
                <FullPanel key={cell.id}>
                  <TransactionCell cell={cell} address={address} cellType={CellType.Output} />
                </FullPanel>
              )}
            />
          ) : (
            <div className="transaction_item__output__empty">{i18n.t('transaction.empty_output')}</div>
          )}
        </div>
      </TransactionCellPanel>
      {address && <TransactionIncome income={transaction.income} />}
    </TransactionPanel>
  )
}

export default TransactionItem
