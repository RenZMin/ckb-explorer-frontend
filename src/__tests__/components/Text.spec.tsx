import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { HighLightLink } from '../../components/Text'

describe('Text Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = (
      <BrowserRouter>
        <HighLightLink value="value" to="to" tooltip="tooltip" />
      </BrowserRouter>
    )
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(HighLightLink)).toHaveLength(1)
  })
})
