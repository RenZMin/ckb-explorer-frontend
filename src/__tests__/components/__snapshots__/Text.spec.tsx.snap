// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component shallow renders 1`] = `
<div
  className="sc-gtsrHT eForCY"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="undefined monospace"
    href="/to"
    onClick={[Function]}
  >
    value
  </a>
</div>
`;
